mr-plonky2-circuits/src/query_erc20/api.rs

```diff
@@ -0,0 +1,74 @@
+use super::{
+    block, revelation,
+    state::{self, CircuitInputsInternal},
+    storage,
+};
+use crate::api::{BlockDBCircuitInfo, C, D, F};
+use serde::{Deserialize, Serialize};
+
+use anyhow::Result;
+use plonky2::plonk::circuit_data::CircuitData;
+
+/// L is the number of elements we allow to expose in the result
+pub enum CircuitInput<const L: usize> {
+    /// Input to be provided to generate a proof for the storage tree circuit of query2-erc20
+    Storage(storage::CircuitInput),
+    /// Input to be provided to generate a proof for the sttate tree circuit of query2-erc20
+    State(state::CircuitInput),
+    /// Input to be provided to generate a proof for the block DB circuit of query2-erc20
+    Block(block::CircuitInput),
+    // Input to be provided to generate a proof for the final revelation circuit of query2-erc20
+    Revelation(revelation::RevelationRecursiveInput),
+}
+
+#[derive(Serialize, Deserialize)]
+/// Parameters representing the circuits employed to prove query2
+pub struct PublicParameters<const BLOCK_DB_DEPTH: usize, const L: usize> {
```

Comment:
`L` should be no longer used for this query I think

Reply:
Resolving it since now we need to keep it since the public inputs have to be the same of query2